frontend/test/resolution/testLoopIndexVars.cpp

```diff
@@ -375,90 +378,146 @@ static ParamLoopTester helpTestParam(const char* rangeExpr, const char* extraCod
   const Module* m = parseModule(context, iterText);
 
   const ResolutionResultByPostorderID& rr = resolveModule(context, m->id());
-  ParamCollector pc;
-  ResolvedVisitor<ParamCollector> rv(&rcval, m, pc, rr);
+  ParamCollector<paramtags::ParamTag::IntParam, IntParam> pc;
+  ResolvedVisitor<decltype(pc)> rv(&rcval, m, pc, rr);
   m->traverse(rv);
 
   const ResolvedExpression& re = rr.byAst(m->stmt(0));
   assert(re.type().type() == IntType::get(context, 0));
-  return ParamLoopTester{std::move(pc), {}, {}};
+  return IntParamLoopTester{std::move(pc), {}, {}};
 }
 
-static void testParamFor() {
+// Note: values hardcoded to match 'helpTestIntParam' helper
+struct EnumParamLoopTester {
+  ParamCollector<paramtags::ParamTag::EnumParam, EnumParam> pc;
+  std::map<std::string, int> resolvedIdents;
+  std::multimap<std::string, std::string> resolvedVals;
+
+  void noteExpectedIteration(std::string e) {
```

Comment:
This might be a little cleaner to use if there was an option for a vector of strings:

```cpp
test.noteExpectedIteration({"red", "green", "blue"});
```

But it's already written, so I guess this is really more of a note for the future. YMMV

Reply:
I've adjusted the tests.